src/haxelib/api/Connection.hx

```diff
@@ -417,12 +417,12 @@ class Connection {
 	}
 
 	static function checkDependencies(dependencies:Dependencies) {
-		for (d in dependencies) {
-			final versions:Array<String> = getVersions(ProjectName.ofString(d.name));
-			if (d.version == "")
+		for (name => versionString in dependencies) {
```

Comment:
Wasn't there something that needed to be compilable with Haxe 3, which doesn't have this syntax?

Reply:
The `haxelib.api` package should be fine, it's just the top level `haxelib` package where things have to be compilable with haxe 3. The `keyValueIterator` method for this abstract is safely kept behind conditional compilation.